### DISCUSSION TITLE
an idea to get rid of proxy timeout

### DISCUSSION THREAD

**Discussion by simulot** [simulot/immich-go#60] [Ideas]
the issue #52 comes when the server is under powered regarding the size of the library.
The fist action done by immich-go is to pull the full list of assets (https://immich.app/docs/api/get-all-assets).

This query gets its data from the database through the immich server and the proxy. The immich server build a JSON response il the dataset, and send it back trough the proxy. Client receives data all at once.

My "production" immich runs on a beefy server. I'm using my laptop as a client. 
The request takes 23 secondes for 95971 images.  The size of data is 201Mb.

When I run the query  using ssh to the nas and docker exec psql, I get the result in 12 seconds. 

```sql
SELECT json_build_object(
    'asset', assets."Id"  ,
    'exif', exif.*
)
FROM assets
LEFT JOIN exif ON assets.id = exif."assetId";
```

``` sh
 cat script.sql | ssh root@mynas   docker  exec -i  immich_postgres psql -d immich postgres -t >out.json
```
 
Should I go in that direction?

**Comment by happytaz** [comment]
I think anything that reduces query time is good. 

I was wondering if the number of duplicates also increases the query time?  I had a library of 70K with 10 duplicates that completed the query in a minute.  My library with 130k had 27K duplicates took about 5 minutes to query.

I'm not sure if it matters but I hope the info can help you.

**Reply by simulot** [reply]
I'm impressed to see such large collection to be powered by a PI 4.

The query time depends only on the number of items in the database. 

The time to search the duplicates among the result is quite negligible compared to the query time.
Then, the 27K duplicates are deleted one by one.

In conclusion, data query done directly on the database container:
- doesn't times out
- is significantly faster

So it worth to implement it.
